### DISCUSSION TITLE
new install on pi zero 2w issues...

### DISCUSSION THREAD

**Discussion by hemna** [aluminum-ice/pwnagotchi#135] [General]
I got my raspi zero 2 w up and running with the latest image.  I have ssh'd in to the pi from my mac and I'm seeing errors in dmesg related to the brcmfmac driver

```
[  118.531581] ieee80211 phy0: brcmf_cfg80211_add_iface: iface validation failed: err=-95
[  119.201560] brcmfmac: brcmf_cfg80211_set_power_mgmt: Forcing power management
[  119.201577] brcmfmac: brcmf_cfg80211_set_power_mgmt: power save disabled
[  119.238336] brcmfmac: brcmf_vif_add_validate: Attempt to add a MONITOR interface...
[  119.238352] brcmfmac: brcmf_vif_add_validate: ... there is already a monitor interface, returning EOPNOTSUPP
[  119.238366] ieee80211 phy0: brcmf_cfg80211_add_iface: iface validation failed: err=-95
[  119.957467] brcmfmac: brcmf_cfg80211_set_power_mgmt: Forcing power management
[  119.957481] brcmfmac: brcmf_cfg80211_set_power_mgmt: power save disabled
```

the display is in AI mode, but stuck with the message
"The neural network is ready" 

```
root@Peepis:/usr/local/bin# ifconfig
bnep0: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 172.20.10.6  netmask 255.255.255.0  broadcast 0.0.0.0
        inet6 fe80::ba27:ebff:fecf:14db  prefixlen 64  scopeid 0x20<link>
        inet6 2600:1003:b846:7942:ba27:ebff:fecf:14db  prefixlen 64  scopeid 0x0<global>
        ether b8:27:eb:cf:14:db  txqueuelen 1000  (Ethernet)
        RX packets 76  bytes 9571 (9.3 KiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 54  bytes 5453 (5.3 KiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

lo: flags=73<UP,LOOPBACK,RUNNING>  mtu 65536
        inet 127.0.0.1  netmask 255.0.0.0
        inet6 ::1  prefixlen 128  scopeid 0x10<host>
        loop  txqueuelen 1000  (Local Loopback)
        RX packets 14277  bytes 70725919 (67.4 MiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 14277  bytes 70725919 (67.4 MiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

mon0: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        unspec D8-3A-DD-7A-7B-6F-30-30-00-00-00-00-00-00-00-00  txqueuelen 1000  (UNSPEC)
        RX packets 11170  bytes 2328712 (2.2 MiB)
        RX errors 0  dropped 115  overruns 0  frame 0
        TX packets 0  bytes 0 (0.0 B)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

usb0: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 10.0.0.2  netmask 255.255.255.0  broadcast 10.0.0.255
        inet6 fe80::5881:4fff:fe76:3ff9  prefixlen 64  scopeid 0x20<link>
        ether 5a:81:4f:76:3f:f9  txqueuelen 1000  (Ethernet)
        RX packets 996  bytes 78817 (76.9 KiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 1341  bytes 295465 (288.5 KiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0
        
```

```
root@Peepis:/usr/local/bin# iwconfig
lo        no wireless extensions.

usb0      no wireless extensions.

wlan0     IEEE 802.11  ESSID:off/any
          Mode:Managed  Access Point: Not-Associated
          Retry short limit:7   RTS thr:off   Fragment thr:off
          Encryption key:off
          Power Management:off

mon0      IEEE 802.11  Mode:Monitor  Frequency:2.412 GHz  Tx-Power=31 dBm
          Retry short limit:7   RTS thr:off   Fragment thr:off
          Power Management:on

bnep0     no wireless extensions.
```

```
root@Peepis:/usr/local/bin# journalctl -fu bettercap
-- Logs begin at Sat 2024-09-07 17:55:00 EDT. --
Sep 07 18:01:15 Peepis bettercap-launcher[435]: command failed: Operation not supported (-95)
Sep 07 18:01:16 Peepis rfkill[6851]: unblock set for all
Sep 07 18:01:16 Peepis bettercap-launcher[435]: command failed: Operation not supported (-95)
Sep 07 18:01:17 Peepis bettercap-launcher[435]: command failed: Operation not supported (-95)
Sep 07 18:01:18 Peepis bettercap-launcher[435]: command failed: Operation not supported (-95)
Sep 07 18:01:18 Peepis bettercap-launcher[435]: command failed: Operation not supported (-95)
Sep 07 18:01:19 Peepis rfkill[6903]: unblock set for all
Sep 07 18:01:19 Peepis bettercap-launcher[435]: command failed: Operation not supported (-95)
Sep 07 18:01:19 Peepis rfkill[6916]: unblock set for all
Sep 07 18:01:20 Peepis bettercap-launcher[435]: command failed: Operation not supported (-95)
Sep 07 18:01:20 Peepis rfkill[6930]: unblock set for all
Sep 07 18:01:21 Peepis bettercap-launcher[435]: command failed: Operation not supported (-95)
Sep 07 18:01:21 Peepis rfkill[6950]: unblock set for all
Sep 07 18:01:21 Peepis bettercap-launcher[435]: command failed: Operation not supported (-95)
Sep 07 18:01:22 Peepis bettercap-launcher[435]: command failed: Operation not supported (-95)
Sep 07 18:01:23 Peepis bettercap-launcher[435]: command failed: Operation not supported (-95)
```

**Comment by wpa-2** [comment]
> I got my raspi zero 2 w up and running with the latest image. I have ssh'd in to the pi from my mac and I'm seeing errors in dmesg related to the brcmfmac driver
> 
> ```
> [  118.531581] ieee80211 phy0: brcmf_cfg80211_add_iface: iface validation failed: err=-95
> [  119.201560] brcmfmac: brcmf_cfg80211_set_power_mgmt: Forcing power management
> [  119.201577] brcmfmac: brcmf_cfg80211_set_power_mgmt: power save disabled
> [  119.238336] brcmfmac: brcmf_vif_add_validate: Attempt to add a MONITOR interface...
> [  119.238352] brcmfmac: brcmf_vif_add_validate: ... there is already a monitor interface, returning EOPNOTSUPP
> [  119.238366] ieee80211 phy0: brcmf_cfg80211_add_iface: iface validation failed: err=-95
> [  119.957467] brcmfmac: brcmf_cfg80211_set_power_mgmt: Forcing power management
> [  119.957481] brcmfmac: brcmf_cfg80211_set_power_mgmt: power save disabled
> ```
> 
> the display is in AI mode, but stuck with the message "The neural network is ready"
> 
> ```
> root@Peepis:/usr/local/bin# ifconfig
> bnep0: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
>         inet 172.20.10.6  netmask 255.255.255.0  broadcast 0.0.0.0
>         inet6 fe80::ba27:ebff:fecf:14db  prefixlen 64  scopeid 0x20<link>
>         inet6 2600:1003:b846:7942:ba27:ebff:fecf:14db  prefixlen 64  scopeid 0x0<global>
>         ether b8:27:eb:cf:14:db  txqueuelen 1000  (Ethernet)
>         RX packets 76  bytes 9571 (9.3 KiB)
>         RX errors 0  dropped 0  overruns 0  frame 0
>         TX packets 54  bytes 5453 (5.3 KiB)
>         TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0
> 
> lo: flags=73<UP,LOOPBACK,RUNNING>  mtu 65536
>         inet 127.0.0.1  netmask 255.0.0.0
>         inet6 ::1  prefixlen 128  scopeid 0x10<host>
>         loop  txqueuelen 1000  (Local Loopback)
>         RX packets 14277  bytes 70725919 (67.4 MiB)
>         RX errors 0  dropped 0  overruns 0  frame 0
>         TX packets 14277  bytes 70725919 (67.4 MiB)
>         TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0
> 
> mon0: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
>         unspec D8-3A-DD-7A-7B-6F-30-30-00-00-00-00-00-00-00-00  txqueuelen 1000  (UNSPEC)
>         RX packets 11170  bytes 2328712 (2.2 MiB)
>         RX errors 0  dropped 115  overruns 0  frame 0
>         TX packets 0  bytes 0 (0.0 B)
>         TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0
> 
> usb0: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
>         inet 10.0.0.2  netmask 255.255.255.0  broadcast 10.0.0.255
>         inet6 fe80::5881:4fff:fe76:3ff9  prefixlen 64  scopeid 0x20<link>
>         ether 5a:81:4f:76:3f:f9  txqueuelen 1000  (Ethernet)
>         RX packets 996  bytes 78817 (76.9 KiB)
>         RX errors 0  dropped 0  overruns 0  frame 0
>         TX packets 1341  bytes 295465 (288.5 KiB)
>         TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0
>         
> ```
> 
> ```
> root@Peepis:/usr/local/bin# iwconfig
> lo        no wireless extensions.
> 
> usb0      no wireless extensions.
> 
> wlan0     IEEE 802.11  ESSID:off/any
>           Mode:Managed  Access Point: Not-Associated
>           Retry short limit:7   RTS thr:off   Fragment thr:off
>           Encryption key:off
>           Power Management:off
> 
> mon0      IEEE 802.11  Mode:Monitor  Frequency:2.412 GHz  Tx-Power=31 dBm
>           Retry short limit:7   RTS thr:off   Fragment thr:off
>           Power Management:on
> 
> bnep0     no wireless extensions.
> ```
> 
> ```
> root@Peepis:/usr/local/bin# journalctl -fu bettercap
> -- Logs begin at Sat 2024-09-07 17:55:00 EDT. --
> Sep 07 18:01:15 Peepis bettercap-launcher[435]: command failed: Operation not supported (-95)
> Sep 07 18:01:16 Peepis rfkill[6851]: unblock set for all
> Sep 07 18:01:16 Peepis bettercap-launcher[435]: command failed: Operation not supported (-95)
> Sep 07 18:01:17 Peepis bettercap-launcher[435]: command failed: Operation not supported (-95)
> Sep 07 18:01:18 Peepis bettercap-launcher[435]: command failed: Operation not supported (-95)
> Sep 07 18:01:18 Peepis bettercap-launcher[435]: command failed: Operation not supported (-95)
> Sep 07 18:01:19 Peepis rfkill[6903]: unblock set for all
> Sep 07 18:01:19 Peepis bettercap-launcher[435]: command failed: Operation not supported (-95)
> Sep 07 18:01:19 Peepis rfkill[6916]: unblock set for all
> Sep 07 18:01:20 Peepis bettercap-launcher[435]: command failed: Operation not supported (-95)
> Sep 07 18:01:20 Peepis rfkill[6930]: unblock set for all
> Sep 07 18:01:21 Peepis bettercap-launcher[435]: command failed: Operation not supported (-95)
> Sep 07 18:01:21 Peepis rfkill[6950]: unblock set for all
> Sep 07 18:01:21 Peepis bettercap-launcher[435]: command failed: Operation not supported (-95)
> Sep 07 18:01:22 Peepis bettercap-launcher[435]: command failed: Operation not supported (-95)
> Sep 07 18:01:23 Peepis bettercap-launcher[435]: command failed: Operation not supported (-95)
> ```

This is probably a little more updated 
https://github.com/jayofelony/pwnagotchi
